pkg/archive/archive.go

```diff
@@ -0,0 +1,224 @@
+package archive
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/mholt/archiver/v3"
+	"github.com/sirupsen/logrus"
+)
+
+type Archiver interface {
+	String() string
+	Archive([]string, string) error
+	Unarchive(string, string) error
+	Write(archiver.File) error
+	Create(io.Writer) error
+	Close() error
+	Walk(string, archiver.WalkFunc) error
+	Open(io.Reader, int64) error
+	Read() (archiver.File, error)
+}
+
+// NewArchiver create a new archiver for tar archive manipultation
+func NewArchiver() Archiver {
+	// Create tar specifically with overrite on false
+	return &archiver.TarGz{
+
+		Tar: &archiver.Tar{
+			OverwriteExisting:      true,
+			MkdirAll:               true,
+			ImplicitTopLevelFolder: false,
+			StripComponents:        0,
+			ContinueOnError:        false,
+		},
+	}
+}
+
+// CreateSplitAchrive will create multiple tar archives from source directory
+func CreateSplitArchive(a Archiver, maxSplitSize int64, destDir, sourceDir, prefix string, newFiles []string) error {
+
+	logrus.Debugf("Found new files %v", newFiles)
+
+	// Declare split variables
+	splitNum := 0
+	splitSize := int64(0)
+	splitPath := fmt.Sprintf("%s/%s_%06d.%s", destDir, prefix, splitNum, a.String())
+
+	splitFile, err := os.Create(splitPath)
+
+	if err != nil {
+		return fmt.Errorf("creating %s: %v", splitPath, err)
+	}
+
+	// Create a new tar archive for writing
+	logrus.Infof("Creating archive %s", splitPath)
+	if a.Create(splitFile); err != nil {
+		return fmt.Errorf("creating archive %s: %v", splitPath, err)
+	}
+
+	sourceInfo, err := os.Stat(sourceDir)
+
+	if err != nil {
+		return fmt.Errorf("%s: stat: %v", sourceDir, err)
+	}
+
+	foundFiles := make(map[string]struct{}, len(newFiles))
+	for _, fpath := range newFiles {
+		foundFiles[fpath] = struct{}{}
+	}
+
+	err = filepath.Walk(sourceDir, func(fpath string, info os.FileInfo, err error) error {
+
+		if err != nil {
+			return fmt.Errorf("traversing %s: %v", fpath, err)
+		}
+		if info == nil {
+			return fmt.Errorf("no file info")
+		}
+
+		if _, found := foundFiles[fpath]; found {
```

Comment:
```suggestion
		if _, found := foundFiles[fpath]; !found {
```

Reply:
@estroz This is set to only process found files, so found should equal true here. Kind of opposite logic from the ReconcileFiles logic; `bundle.ReconcileFiles` now returns a slice of string with the files names that should be packed.

Reply:
SGTM. The log message below threw me off, thanks for clarifying.